Toolkit/ArcGISToolkit/TimeSlider.swift

```diff
@@ -0,0 +1,2539 @@
+            var endTimeLabelX = upperThumbLayer.frame.midX - endTimeLabelSize.width / 2
+            currentExtentEndTimeLabel.isHidden = !isSliderVisible
+            if endTimeLabelX < bounds.origin.x + labelSidePadding && currentExtentStartTime == currentExtentEndTime {
+                endTimeLabelX = bounds.origin.x + labelSidePadding
+                currentExtentEndTimeLabel.isHidden = true
+            }
+            else if endTimeLabelX + endTimeLabelSize.width > bounds.maxX - labelSidePadding {
+                endTimeLabelX = bounds.maxX - endTimeLabelSize.width - labelSidePadding
+                currentExtentEndTimeLabel.isHidden = true
+            }
+            else if endTimeLabelX < currentExtentStartTimeLabel.frame.origin.x + currentExtentStartTimeLabel.frame.width {
+                endTimeLabelX = currentExtentStartTimeLabel.frame.origin.x + currentExtentStartTimeLabel.frame.width + paddingBetweenLabels
+            }
+            
+            let thumbEndTimeLabelY = upperThumbLayer.frame.minY - currentExtentEndTimeLabel.frame.height - labelPadding
+            let tickLayerEndTimeLabelY = tickMarkLayer.frame.minY - currentExtentEndTimeLabel.frame.height - labelPadding
+            let endTimeLabelY = min(thumbEndTimeLabelY, tickLayerEndTimeLabelY)
+            currentExtentEndTimeLabel.frame = CGRect(x: endTimeLabelX, y: endTimeLabelY, width: endTimeLabelSize.width, height: endTimeLabelSize.height)
+        }
+        else {
+            currentExtentEndTimeLabel.string = ""
+            currentExtentEndTimeLabel.isHidden = true
+        }
+        
+        // Commit the transaction
+        CATransaction.commit()
+    }
+    
+    private func positionTickMarks() {
+        //
+        // Bail out if time steps are not available
+        guard let timeSteps = timeSteps, timeSteps.count > 0, isSliderVisible else {
+            return
+        }
+        
+        // Create tick marks based on
+        // time steps and it's calculated
+        // position on the slider
+        var tms = [TickMark]()
+        for i in 0..<timeSteps.count  {
+            let timeStep = timeSteps[i]
+            let tickX = CGFloat(position(for: timeStep.timeIntervalSince1970))
+            tms.append(TickMark(originX: tickX, value: timeStep))
+        }
+        
+        // If label mode is ticks then we need to
+        // find out the major/minor ticks and
+        // add labels for major ticks
+        if labelMode == .ticks {
+            var majorTickInterval = 2
+            var doMajorTickLabelsCollide = false
+            var firstMajorTickIndex = 0
+            let tickCount = tms.count
+            
+            // Calculate the largest number of ticks to allow between major ticks. This prevents scenarios where
+            // there are two major ticks placed undesirably close to the end.
+            let maxMajorTickInterval = Int(ceil(Double(tickCount / 2)))
```

Comment:
I think you intended to cast `tickCount` to `Double` before dividing by `2`. The way it is now, both `tickCount` and `2` are integers, so integer division will be used. The value of `maxMajorTickInterval` would just be the quotient.

Reply:
fixed

Reply:
I want result as `Int` so this is fine.

Reply:
Let me break `Int(ceil(Double(tickCount / 2)))` down into steps:
```swift
let step1 = tickCount / 2
let step2 = Double(step1)
let step3 = ceil(step2)
let step4 = Int(step4)
```
Suppose there are `3` ticks. Let's look at each of the steps. 
```swift
let tickCount = 3 // 3 (Int)
let step1 = tickCount / 2 // step1 = 3 (Int) / 2 (Int) = 1 (Int)
let step2 = Double(step1) // step2 = Double(1) = 1.0 (Double)
let step3 = ceil(step2) // step3 = ceil(1.0) = 1.0 (Double)
let step4 = Int(step4) // step4 = Int(1.0) = 1 (Int)
```
Now suppose we cast `tickCount` to `Double` before doing anything else: `Int(ceil(Double(tickCount) / 2))`. Let me break that down into steps:
```swift
Int(ceil(Double(tickCount) / 2))
let step1 = Double(tickCount)
let step2 = step1 / 2
let step3 = ceil(step2)
let step4 = Int(step3)
```
Again, suppose there are `3`ticks:
```swift
let tickCount = 3 // 3 (Int)
let step1 = Double(tickCount) // step1 = Double(3) = 3.0 (Double)
let step2 = step1 / 2 // step2 = 3.0 (Double) / 2 (Double) = 1.5 (Double)
let step3 = ceil(step2) // step3 = ceil(1.5) = 2.0 (Double)
let step4 = Int(step3) // step4 = Int(2.0) = 2 (Int)
```
@njarecha Does that make sense?

Reply:
Yes, make sense now. Fixed